### PR TITLE
Update Java Setup for JDK 17

### PR DESCRIPTION
**Pull request by sbudhirajadoc**
We updated set documentation to include instructions for JDK 17.

### What issues does this PR fix or reference?
#<Insert GitHub Issue>, @https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000TxnSEYAZ/view@

### Functionality Before
Documentation didn't include instructions for JDK 17

### Functionality After
Documentation now includes instructions for JDK 17.

### CODE REVIEW

**Comment by sbudhirajadoc** [comment]
Need help with updating all the salesforcedx-vscode-apex.java.home paths since I cannot test them.

File: docs/_articles/en/getting-started/java-setup.md [hunk 1]
```diff
@@ -86,13 +86,10 @@ Windows:
 ## Oracle Java 11 (Officially Tested)
 
 1. Navigate to the [download page](https://www.oracle.com/technetwork/java/javase/downloads/jdk11-downloads-5066655.html)
```
Comment (jeffb-sfdc) [inline docs/_articles/en/getting-started/java-setup.md:88]: https://www.oracle.com/java/technologies/downloads/#java17

File: docs/_articles/en/getting-started/java-setup.md [hunk 2]
```diff
@@ -86,13 +86,10 @@ Windows:
...
-1. Click the **Download** button for **Java SE 11.x.y (LTS)**
-1. Accept the license agreement
-1. Click the download link that applies to your OS
-1. For macOS download the dmg version.
-1. For Windows Download exe version.
-1. If prompted to log in, use your Oracle account.
-1. Once the file is downloaded open it and complete the installation steps.
+2. Navigate to **Java SE Development Kit 17.0.2 downloads**.
```
Comment (jeffb-sfdc) [inline docs/_articles/en/getting-started/java-setup.md:89]: https://www.oracle.com/java/technologies/downloads/#java17

File: docs/_articles/en/getting-started/java-setup.md [hunk 3]
```diff
@@ -13,15 +13,14 @@ The Apex Language Server, shipped as part of the Salesforce Apex Extension for V
 
 Below you'll find instructions on how to download and configure different versions of Java.
 
-## AdoptOpenJDK
+## Adoptium
 
-[AdoptOpenJDK](https://adoptopenjdk.net/) provides prebuilt OpenJDK binaries for free without authentication or complications.
+[Adoptium](https://adoptopenjdk.net/) provides prebuilt OpenJDK binaries for free without authentication or complications.
```
Comment (jag-sfdc) [inline docs/_articles/en/getting-started/java-setup.md:18]: Do we want to point to adoptium.net link here?

Comment (sbudhirajadoc) [reply docs/_articles/en/getting-started/java-setup.md:18]: good catch. will update

File: docs/_articles/en/getting-started/java-setup.md [hunk 4]
```diff
@@ -13,15 +13,14 @@ The Apex Language Server, shipped as part of the Salesforce Apex Extension for V
...
 
-1. Navigate to [AdoptOpenJDK](https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=hotspot)
-1. Select **OpenJDK 11 (LTS)**
-1. Select **HotSpot**
-1. Click the **Latest Release** button to download.
-1. After the file has downloaded, open it and complete the installation steps.
+1. Navigate to [Adoptium](https://adoptium.net/?variant=openjdk11)
+2. Select **Temurin 17 (LTS)**
```
[the remark this answers is not in the document]
Comment (sbudhirajadoc) [reply docs/_articles/en/getting-started/java-setup.md:21]: I saw them occasionally rendering incorrectly and decided to play it safe.

**Review by jag-sfdc — APPROVED** [review]

**Pull request merged**